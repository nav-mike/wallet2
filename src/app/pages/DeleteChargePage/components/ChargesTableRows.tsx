import React from 'react';
import { ICharge } from './ChargesTable';
import { ChargesTableItem } from './ChargesTableItem';

export function ChargesTableRows(props: { charges: ICharge[] }) {
  const data = props.charges.map((item: ICharge) => (
    <ChargesTableItem item={item} />
  ));
  return <tbody>{data}</tbody>;
}
