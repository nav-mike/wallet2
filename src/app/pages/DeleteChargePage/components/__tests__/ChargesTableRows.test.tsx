import * as React from 'react';
import { render } from '@testing-library/react';
import { ChargesTableRows } from '../ChargesTableRows';

describe('<ChargesTableRows />', () => {
  const data = [
    { id: '123-123', date: new Date(), value: 120.21, type: 'food' },
  ];

  it('should render and match the snapshot', () => {
    const rows = render(<ChargesTableRows charges={data} />, {
      container: document.createElement('table'),
    });
    expect(rows.container.firstChild).toMatchSnapshot();
  });
});
