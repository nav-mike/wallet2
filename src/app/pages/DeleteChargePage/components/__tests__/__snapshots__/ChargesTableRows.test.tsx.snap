// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChargesTableRows /> should render and match the snapshot 1`] = `
.c0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  background-color: rgba(220,120,95,1);
  border: 1px solid rgba(220,120,95,1);
  border-radius: 0.5rem;
  min-height: 2rem;
}

.c0:hover {
  cursor: pointer;
  background-color: #fab9a9;
}

<tbody>
  <tr>
    <td>
      3/17/2021
    </td>
    <td>
      120.21
    </td>
    <td>
      food
    </td>
    <td>
      <button
        class="c0"
      >
        Delete
      </button>
    </td>
  </tr>
</tbody>
`;
